pkg/apis/rke.cattle.io/v1/cluster.go

```diff
@@ -88,6 +88,8 @@ type DrainOptions struct {
 	Timeout int `json:"timeout"`
 	// SkipWaitForDeleteTimeoutSeconds If pod DeletionTimestamp older than N seconds, skip waiting for the pod.  Seconds must be greater than 0 to skip.
 	SkipWaitForDeleteTimeoutSeconds int `json:"skipWaitForDeleteTimeoutSeconds"`
+	// PodSelector for selecting pods to drain
+	PodSelector string `json:"podSelector"`
```

Comment:
On the SUC side this is a full `metav1.LabelSelector`: 
https://github.com/rancher/system-upgrade-controller/blob/ac25caa9da7f6ab5ce6298cc8e2e0bc7e8b120eb/pkg/apis/upgrade.cattle.io/v1/types.go#L86

Will you also include logic to convert the string to the correct type for use in the Plan? There are plenty of helpers to do so in the Kubernetes codebase, I was just concerned that you were going to try to feed this string directly into the Plan's DrainSpec.

Reply:
I added `labels.Parse` to validate users' input. In the harvester, we modify `rkeConfig` in the provisioning cluster to upgrade k8s, so we may still need `PodSelector` in `DrainOptions`.